src/api/useRequest/index.ts

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+import axios from 'axios';
+import { pathOr, cond, T, defaultTo } from 'ramda';
+
+import { useCancelTokenSource, Severity } from '../..';
+import useSnackbar from '../../Snackbar/useSnackbar';
+
+export interface RequestParams<TResult> {
+  request: (token) => Promise<TResult>;
+  getErrorMessage?: (error) => string;
+  defaultFailureMessage?: string;
+}
+
+export interface RequestResult<TResult> {
+  sendRequest: () => Promise<TResult>;
+  sending: boolean;
+}
+
+const useRequest = <TResult>({
+  request,
+  getErrorMessage,
+  defaultFailureMessage = 'Oops, something went wrong',
+}: RequestParams<TResult>): RequestResult<TResult> => {
+  const { token, cancel } = useCancelTokenSource();
+  const { showMessage } = useSnackbar();
+
+  const [sending, setSending] = React.useState(true);
```

Comment:
sending should be true only when sendRequest is called
```suggestion
  const [sending, setSending] = React.useState(false);
```